src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceGeneratedFileInfo.cs

```diff
@@ -21,16 +22,24 @@ internal sealed class MetadataAsSourceGeneratedFileInfo
 
         private readonly ParseOptions _parseOptions;
 
-        public MetadataAsSourceGeneratedFileInfo(string rootPath, Project sourceProject, INamedTypeSymbol topLevelNamedType)
+        public MetadataAsSourceGeneratedFileInfo(string rootPath, Project sourceProject, INamedTypeSymbol topLevelNamedType, bool allowDecompilation)
         {
             this.SourceProjectId = sourceProject.Id;
-            _parseOptions = sourceProject.ParseOptions;
             this.Workspace = sourceProject.Solution.Workspace;
-            this.LanguageName = sourceProject.Language;
+            this.LanguageName = allowDecompilation ? LanguageNames.CSharp : sourceProject.Language;
+            if (sourceProject.Language == LanguageName)
+            {
+                _parseOptions = sourceProject.ParseOptions;
+            }
+            else
+            {
+                _parseOptions = Workspace.Services.GetLanguageServices(LanguageName).GetRequiredService<ISyntaxTreeFactoryService>().GetDefaultParseOptionsWithLatestLanguageVersion();
+            }
```

Comment:
nit: consider ?:

Reply:
Opted against it here. The two branches seem different enough that I like to keep it obvious.

Reply:
Why still having the branches at all? When would the source project's parse options be different than the default parse options for the language in an observable way?

I'm not saying there's not a reason for this....I just can't tell what it is. I can't read your mind (not sure if good thing or bad thing?) but do we need a comment here?